bypass/bypass.go

```diff
@@ -84,24 +84,43 @@ func (b *bypass) OnProxies(infos map[string]*commonconfig.ProxyConfig, configDir
 	}
 
 	dialers := chained.CreateDialersMap(configDir, supportedInfos, userConfig)
-	for k, v := range supportedInfos {
-		dialer := dialers[k]
+	for name, config := range supportedInfos {
+		dialer := dialers[name]
 		if dialer == nil {
-			log.Errorf("No dialer for %v", k)
+			log.Errorf("No dialer for %v", name)
 			continue
 		}
 
-		pc := chained.CopyConfig(v)
-		// Set the name in the info since we know it here.
-		pc.Name = k
-		// Kill the cert to avoid it taking up unnecessary space.
-		pc.Cert = ""
-		p := b.newProxy(k, pc, configDir, userConfig, dialer)
-		b.proxies = append(b.proxies, p)
-		go p.start()
+		// if dialer is not ready, try again in 60s
+		if !dialer.IsReady() {
+			log.Debugf("dialer %q is not ready, starting in background", name)
+			go func() {
+				for {
+					time.Sleep(60 * time.Second)
+					if dialer.IsReady() {
+						b.startProxy(name, config, configDir, userConfig, dialer)
+						break
+					}
+				}
+			}()
```

Comment:
Hmmmmm thanks for the suggestion! I'll make IsReady return an error too, so if err is nil and ok is false then it should try again.

Reply:
In this way we can also customize when we can also retry to start a proxy or call the func to create a implementation again depending of the returned error

Reply:
This will still suffer from the same problem though; It's not guaranteed that `isReady` will eventually return an it's ready or an error. If, at some point, we were to update `water` to retry downloading the WASM when it fails, or add a new protocol that might not be ready immediately, the caller can't assume that the dialer itself isn't running an infinite loop. Or there might be a bug where `isReady` never returns true or a non nil error. The goroutines won't ever be cleaned up, leading to memory leaks that we may not catch.

Reply:
Hmmmm it makes sense, thanks catching this! I've updated the bypass code:

1. Added a go routine function `loadProxyAsync` for creating a context that timeout after 10 min.
2. The function also provides an atomic boolean for the internal go routine that actually check if the proxy is ready or not. So the `loadProxyAsync` waits until the context timeout or it receives a channel signal saying it's ready. When ready the internal go routine should stop looping with the flag marked as false and the ready channel is closed.